mle/agents/clarifier.py

```diff
@@ -0,0 +1,76 @@
+from rich.console import Console
+
+from mle.function import *
+
+
+class ClarifierAgent:
```

Comment:
We don't need to build a new agent, please add a new function in the advisor agent to recommend a specific dataset when the user's dataset name is blurred. Provide selection options instead of calling functions to ask user

Reply:
Thanks for your suggestions, @huangyz0918!

I tried to implement your suggestions, but faced two barriers:

1. Do you mean we should provide the suggested dataset lists at the adviser calling stage (step 3), or call the adviser to suggest the dataset list immediately after the user inputs the blur dataset (step 1)?

2. if we don't use function calls to provide the dataset lists, we'll need the dataset lists to be direct LLM output. This requires a fixed JSON output example in the adviser's system prompt. However, the adviser's system prompt already has a fixed JSON example, which would create a conflict. 😂

If you have time, could you please share more ideas on how to incorporate this dataset suggestion into the adviser's workflow? (Hope to align myself more with you). I'll implement it tmr soon. Thanks a lot!!

Reply:
> Do you mean we should provide the suggested dataset lists at the adviser calling stage (step 3), or call the adviser to suggest the dataset list immediately after the user inputs the blur dataset (step 1)?

Step 1. -- There are many functions in the advisor agent, chat/interact/suggest, etc. Some of them are sharing the same chat history, others are not. This dataset recommendation should use a standalone chat history to complete a simple task. 

> if we don't use function calls to provide the dataset lists, we'll need the dataset lists to be direct LLM output. This requires a fixed JSON output example in the adviser's system prompt. However, the adviser's system prompt already has a fixed JSON example, which would create a conflict. 😂

Think differently, it is a two-step progress: first, check the user's input is blurred or not (output is a boolean) -> if yes, generate several options for the user to select, and then use the selected one for further advisor suggestion. 

You should add a single function in the advisor agent, but the function is not directly related to the main suggestion progress of the advisor agent. the function should be called in the workflow, for example:

```python
blur_dataset = ask_text("Please provide your dataset information (a public dataset name or a local file path)")

# make sure the dataset is a path to a local file or a specific name
dataset = advisor.clarify_dataset(blur_dataset)

# suggestions = advisor.interact(...)
```

There is a function called `ask_data` in the `baseline.py`, you should merge the `ask_text("Please provide your dataset information (a public dataset name or a local file path)")` into `ask_data` and then call the advisor agent in this function.